hazelcast/src/main/java/com/hazelcast/client/impl/spi/ProxyManager.java

```diff
@@ -268,7 +268,7 @@ public void register(final String serviceName, final Class<? extends ClientProxy
     }
 
     public ClientProxy getOrCreateProxy(@Nonnull String service,
-                                        @Nonnull String id) {
+                                        @Nonnull String id, boolean remote) {
```

Comment:
Instead of using such boolean to differentiate the local and remote, can you define another methods like `getOrCreateLocalProxy` and `getOrCreateProxy`. This way it is more clear. You can always reuse common code. See EventService for registerListener and registerLocalListener.

Reply:
Will add that